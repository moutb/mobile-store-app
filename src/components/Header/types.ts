export type CartIconProps = {
    filled: boolean;
};

export type HeaderProps = {
    showCart?: boolean;
    clear?: boolean;
    title?: string;
};
