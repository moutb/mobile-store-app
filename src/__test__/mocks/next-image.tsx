jest.mock('next/image', () => ({
    __esModule: true,
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    default: (props: any) => {
        // Remove priority prop for snapshot testing
        delete props.priority;
        return <img {...props} />;
    },
}));
