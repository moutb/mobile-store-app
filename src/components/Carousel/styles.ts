import styled from 'styled-components';

export const Wrapper = styled.div`
    display: flex;
    align-items: center;
    flex-flow: column nowrap;
    gap: 0.5rem;
    width: 100%;
`;

export const NavButton = styled.button`
    background: none;
    border: none;
    font-size: var(--font-size-md);
    cursor: pointer;
    color: var(--color-primary);
    padding: 0 0.5rem;

    &:disabled {
        opacity: 0.4;
        cursor: not-allowed;
    }

    &:focus {
        outline: 2px solid currentColor;
    }
`;

export const CarouselTrack = styled.div`
    display: flex;
    overflow-x: hidden;
    scroll-behavior: smooth;
    scroll-snap-type: x mandatory;
    flex: 1;
    width: 100%;
    user-select: none;
`;

export const CarouselItem = styled.div<{ $visibleCount: number }>`
    flex: 0 0 calc(100% / ${({ $visibleCount }) => $visibleCount});
    scroll-snap-align: start;
`;

export const ProgressBar = styled.div`
    width: 100%;
    height: 1px;
    background-color: var(--color-secondary);
    margin-top: var(--spacing-6);
    position: relative;
    border-radius: 2px;
    cursor: pointer;
    user-select: none;
`;

export const ScrollThumb = styled.div`
    position: absolute;
    height: 100%;
    background-color: var(--color-primary);
    border-radius: 2px;
    transition: left 0.2s ease;
`;
