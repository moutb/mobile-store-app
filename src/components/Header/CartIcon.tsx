import Image from 'next/image';
import { CartIconProps } from './types';
import styled from 'styled-components';

const CartIcon = ({ filled }: CartIconProps) => (
    <StyledImage
        src={`/bag${filled ? '-filled' : ''}.svg`}
        alt="Shopping cart"
        width={filled ? 32 : 24}
        height={filled ? 32 : 24}
        style={{ display: 'block' }}
    />
);

const StyledImage = styled(Image)`
    display: block;
`;

export default CartIcon;
