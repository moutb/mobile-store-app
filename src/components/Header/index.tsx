import Link from 'next/link';
import styled from 'styled-components';
import { useCart } from '@/features/cart/context/CartContext';
import Flexbox from '../Flexbox';
import Container from '../Container';
import Image from 'next/image';
import CarIcon from './CartIcon';

const Header = () => {
    const { state } = useCart();
    const totalItems = state.products.length;

    return (
        <HeaderContainer as="header" role="banner">
            <Flexbox as="nav" aria-label="Main navigation">
                <Link href="/" aria-label="Navigate to home page">
                    <Image src="/mbst.svg" alt="Logo" width={74} height={24} />
                </Link>
                <Cart href="/cart" aria-label="Navigate to shopping cart">
                    <CarIcon filled={totalItems > 0} />
                    <CartNumber>{totalItems}</CartNumber>
                </Cart>
            </Flexbox>
        </HeaderContainer>
    );
};

export default Header;

const HeaderContainer = styled(Container)`
    height: var(--header-height);
    padding-top: var(--spacing-4);
    padding-bottom: var(--spacing-4);
`;

const Cart = styled(Link)`
    padding: var(--spacing-1) 0;
    font-size: ${({ theme }) => theme.typography.fontSize.base};
    display: flex;
    align-items: center;
    flex-direction: row;
    gap: 6px;
`;

const CartNumber = styled.span`
    font-size: var(--font-size-base);
    line-height: var(--font-size-base);
    font-weight: var(--font-weight-thin);
    color: var(--color-text);
    text-transform: uppercase;
    flex-grow: 0;
    /* small adjust because of Helvetica numbers Typography issue */
    transform: translateY(2px);
`;
