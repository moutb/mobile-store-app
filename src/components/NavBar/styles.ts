import Image from 'next/image';
import styled from 'styled-components';

export const Nav = styled.nav`
    background-color: ${({ theme }) => theme.colors.background};
    padding: var(--spacing-4) 0;
`;

export const BackButton = styled.button`
    display: inline-flex;
    align-items: center;
    gap: var(--spacing-1);
    background: none;
    border: none;
    font-size: var(--font-size-sm);
    line-height: var(--font-size-sm);
    font-weight: var(--font-weight-thin);
    color: var(--color-text);
    cursor: pointer;
    text-transform: uppercase;

    &:focus {
        outline: 2px solid var(--color-primary);
        outline-offset: 2px;
    }
`;

export const Icon = styled(Image)`
    flex-shrink: 0;
`;
