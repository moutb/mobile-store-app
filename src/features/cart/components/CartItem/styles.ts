import { media } from '@/styles/media';
import Image from 'next/image';
import styled from 'styled-components';
import { css } from 'styled-components';

export const ItemContainer = styled.div`
    display: flex;
    gap: var(--spacing-4);
    align-items: center;
    padding-bottom: var(--spacing-4);
`;

export const ProductImageWrapper = styled.div`
    position: relative;
    height: var(--product-image-wrapper-height);
    width: 40%;

    ${media('md')(css`
        height: var(--product-image-wrapper-md-width);
        width: var(--product-image-wrapper-md-width);
    `)}
`;

export const ProductImage = styled(Image)`
    flex-shrink: 0;
`;

export const Info = styled.div`
    flex: 1;
    display: flex;
    flex-direction: column;
    gap: var(--spacing-2);
    height: 100%;
    max-height: var(--product-info-max-height);
`;

const ProductTextInfo = css`
    font-size: var(--font-size-sm);
    font-weight: var(--font-weight-thin);
    text-transform: uppercase;
`;

export const Name = styled.h2`
    ${ProductTextInfo}
`;

export const Description = styled.p`
    ${ProductTextInfo}
`;

export const Price = styled.p`
    ${ProductTextInfo}
    margin-top: var(--spacing-4);
`;

export const Spacer = styled.div`
    flex-grow: 1;
    flex-shrink: 0;
    width: 1px;
`;

export const RemoveButton = styled.button`
    align-self: flex-start;
    background: none;
    border: none;
    padding: 0;
    color: var(--color-danger);
    font-size: var(--font-size-sm);
    cursor: pointer;

    &:hover {
        text-decoration: underline;
    }

    &:focus-visible {
        outline: 2px solid var(--color-primary);
        outline-offset: 2px;
    }
`;
