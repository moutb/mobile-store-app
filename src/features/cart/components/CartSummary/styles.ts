// components/cart/styles/CartSummary.styles.ts
import { media } from '@/styles/media';
import Link from 'next/link';
import styled from 'styled-components';
import { css } from 'styled-components';

export const SummaryContainer = styled.div`
    display: flex;
    flex-direction: column;
    gap: var(--spacing-4);
    margin-top: var(--spacing-6);
    padding-bottom: var(--spacing-2);
`;

export const Actions = styled.div`
    display: flex;
    gap: var(--spacing-2);
    flex-wrap: wrap;
    width: 100%;
    justify-content: space-between;
    align-items: center;

    ${media('md')(css`
        flex-wrap: nowrap;
        justify-content: flex-end;
        gap: var(--spacing-6);
    `)}
`;

export const TotalRow = styled.div`
    display: flex;
    justify-content: space-between;
    width: 100%;
    gap: var(--spacing-4);
    font-size: var(--font-size-sm);
    font-weight: var(--font-weight-thin);

    ${media('md')(css`
        justify-content: flex-end;
    `)}
`;

export const TotalLabel = styled.span`
    color: var(--color-text-secondary);
`;

export const TotalPrice = styled.span`
    color: var(--color-text);
`;

const ActionButton = css`
    padding: var(--spacing-4) 0;
    cursor: pointer;
    font-size: var(--font-size-xs);
    line-height: var(--font-size-xs);
    font-weight: var(--font-weight-thin);
    text-transform: uppercase;
    width: calc(50% - var(--spacing-2));
    white-space: nowrap;
    text-align: center;
    margin-top: var(--spacing-4);

    ${media('md')(css`
        max-width: 260px;
        padding: var(--spacing-4) var(--spacing-6);
        font-size: var(--font-size-sm);
        line-height: var(--font-size-sm);
        margin-top: 0;
    `)}
`;

export const OutlinedButton = styled(Link)`
    ${ActionButton}
    background: none;
    color: var(--color-text);
    border-width: 0.5px;
    border-style: solid;
    border-color: var(--color-primary);

    &:hover {
        background: var(--color-secondary);
    }

    &:focus-visible {
        outline: 2px solid var(--color-primary);
        outline-offset: 2px;
    }

    ${media('md')(css`
        order: -1;
    `)}
`;

export const FilledButton = styled.button`
    ${ActionButton}
    background: var(--color-primary);
    color: var(--color-white);
    border: none;

    &:hover {
        opacity: 0.9;
    }

    &:focus-visible {
        outline: 2px solid var(--color-primary);
        outline-offset: 2px;
    }
`;
