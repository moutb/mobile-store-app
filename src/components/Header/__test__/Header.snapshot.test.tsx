/* eslint-disable @next/next/no-img-element, jsx-a11y/alt-text */
import { render } from '@testing-library/react';
import Header from '../index';
import { CartProvider } from '@/features/cart/context/CartContext';
import '@testing-library/jest-dom';
import { ThemeProvider } from 'styled-components';
import { theme } from '@/styles/theme';
import '@/__test__/mocks/next-image';

jest.mock('next/navigation', () => ({
    useRouter: () => ({
        push: jest.fn(),
        back: jest.fn(),
    }),
    usePathname: () => '/',
    useSearchParams: () => new URLSearchParams(),
}));

describe('<Header />', () => {
    it('matches snapshot', () => {
        const { asFragment } = render(
            <ThemeProvider theme={theme}>
                <CartProvider initialState={{ products: [] }}>
                    <Header showCart clear />
                </CartProvider>
            </ThemeProvider>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
