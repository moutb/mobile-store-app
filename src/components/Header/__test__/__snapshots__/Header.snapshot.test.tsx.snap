// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> matches snapshot 1`] = `
<DocumentFragment>
  .c2 {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  align-items: center;
  justify-content: space-between;
  gap: 0;
}

.c0 {
  width: 100%;
  margin: 0 auto;
  padding-left: var(--spacing-3);
  padding-right: var(--spacing-3);
  max-width: 100%;
}

.c5 {
  display: block;
}

.c1 {
  height: var(--header-height);
  padding-top: var(--spacing-4);
  padding-bottom: var(--spacing-4);
}

.c1.header--bottom-bordered {
  border-bottom: 1px solid var(--color-primary);
}

.c1>nav {
  height: 100%;
}

.c3 {
  display: block;
  width: 100%;
  height: auto;
}

.c4 {
  font-size: var(--font-size-base);
  display: flex;
  align-items: center;
  flex-direction: row;
  gap: 6px;
}

.c6 {
  font-size: var(--font-size-base);
  line-height: var(--font-size-base);
  font-weight: var(--font-weight-thin);
  color: var(--color-text);
  text-transform: uppercase;
  flex-grow: 0;
  transform: translateY(2px);
}

@media (min-width:360px) {
  .c0 {
    max-width: var(--container-width-xs);
  }
}

@media (min-width:480px) {
  .c0 {
    max-width: var(--container-width-sm);
  }
}

@media (min-width:768px) {
  .c0 {
    max-width: var(--container-width-md);
  }
}

@media (min-width:1024px) {
  .c0 {
    max-width: var(--container-width-lg);
    padding-left: 100px;
    padding-right: 100px;
  }
}

@media (min-width:1280px) {
  .c0 {
    max-width: var(--container-width-xl);
  }
}

<header
    class="c0 c1"
    role="banner"
  >
    <nav
      aria-label="Main navigation"
      class="c2"
    >
      <a
        aria-label="Navigate to home page"
        href="/"
      >
        <img
          alt="Logo"
          class="c3"
          data-nimg="1"
          decoding="async"
          height="48"
          src="/mbst.svg"
          style="color: transparent;"
          width="148"
        />
      </a>
      <a
        aria-label="Navigate to shopping cart"
        class="c4"
        href="/cart"
      >
        <img
          alt="Shopping cart"
          class="c5"
          data-nimg="1"
          decoding="async"
          height="24"
          loading="lazy"
          src="/bag.svg"
          style="color: transparent; display: block;"
          width="24"
        />
        <span
          class="c6"
        >
          0
        </span>
      </a>
    </nav>
  </header>
</DocumentFragment>
`;
